src/vellum/core/context.py

```diff
@@ -0,0 +1,82 @@
+from contextlib import contextmanager
+from functools import wraps
+import inspect
+import threading
+from typing import Any, Callable, Dict, Iterator, TypeVar, Union, cast
+
+from vellum.workflows.events.types import ParentContext
+
+LoggingContext = Dict[str, Any]
```

Comment:
[nit] This should be `ExecutionContext` correct?

Reply:
Would also be nice stringly type this, instead of us calling `.get("parent_context")` below